src/evolveP.js

```diff
@@ -23,9 +26,10 @@ const evolveP = (transforms, obj) => {
       .then(pair(key))
   }
 
-  return Promise.resolve(toPairs(obj))
-    .then(mapP(transform))
-    .then(fromPairs)
+  return compose(
+    composeP(fromPairs, mapP(transform), resolve),
```

Comment:
Do you really need `resolve`?

Reply:
I'm thinking no, since `mapP` returns a promise.